relay-server/src/actors/processor.rs

```diff
@@ -1013,42 +1014,60 @@ impl EnvelopeProcessorService {
         }
 
         envelope.retain_items(|item| match item.ty() {
-            ItemType::Profile => match relay_profiling::expand_profile(&item.payload()) {
-                Ok(payload) => {
-                    if payload.len() <= self.config.max_profile_size() {
-                        item.set_payload(ContentType::Json, payload);
-                        true
-                    } else {
+            ItemType::Profile => {
+                let contexts = match state.event.value_mut() {
+                    Some(event) if event.ty.value() == Some(&EventType::Transaction) => {
+                        event.contexts.value_mut().as_mut()
+                    }
+                    _ => None,
+                };
+                match relay_profiling::expand_profile(&item.payload()) {
+                    Ok((profile_id, payload)) => {
+                        if payload.len() <= self.config.max_profile_size() {
+                            if let Some(contexts) = contexts {
+                                contexts.add(SentryContext::Profile(Box::new(ProfileContext {
+                                    profile_id: Annotated::new(profile_id),
+                                })));
```

Comment:
This overwrites the context if it was already set by an SDK. I assume this is intentional?

Reply:
This is indeed intentional.